src/modularity.jl

```diff
@@ -0,0 +1,63 @@
+"""
+    randompartitioning(hg::Hypergraph, n::Int)::Vector{Vector{Int}}
+    
+Generates a random `n`-partitioning for a hypergraph `hg`. 
+"""
+function randompartitioning(hg::Hypergraph, n::Int)::Vector{Vector{Int}}
+    x = rand(1:n, size(hg, 1))
+    [findall(==(i), x) for i in 1:n]
+end
+
+"""
+    modularity(hg::Hypergraph, partitioning::Vector{Vector{Int}})
+    
+Calculates the modularity of a hypergraph `hg` for a given `partitioning`.
+"""
+function modularity(hg::Hypergraph, partitioning::Vector{Vector{Int}})
+    @assert sum(length.(partitioning)) == size(hg, 1)
```

Comment:
it should be possible to disable these tests as this is a performance critical function

Reply:
what patter you want to use here? do you want to add parameter perform_checks=true ?

Reply:
We can use `@inbounds`/`@boundscheck` mechanics. It is not exactly the same as for arrays but mentally similar. Then the function needs to be inlined.

Reply:
added `@inline` and `@boundscheck`